rustls/src/client/common.rs

```diff
@@ -27,12 +27,16 @@ impl ServerCertDetails {
 
 pub(super) struct ClientHelloDetails {
     pub(super) sent_extensions: Vec<ExtensionType>,
+    pub(super) extension_order_seed: u32,
 }
 
 impl ClientHelloDetails {
-    pub(super) fn new() -> Self {
+    pub(super) fn new(mut extension_order_seed: u32) -> Self {
+        // leave bottom 16 bits clear for extension type.
+        extension_order_seed &= 0xffff0000;
         Self {
             sent_extensions: Vec::new(),
+            extension_order_seed,
```

Comment:
Would prefer to instead write:

```
            // Leave bottom 16 bits clear for extension type.
            extension_order_seed: extension_order_seed & 0xffff_0000,
```

(That is, avoiding the extra binding.)

Reply:
Since I'm now generating u16 this is changed a bit